frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerThread.java

```diff
@@ -162,13 +168,13 @@ public void setMemory(long memory) {
         this.memory = memory;
     }
 
-    public void connect() throws WorkerInitializationException {
+    private void connect() throws WorkerInitializationException {
         if (!configManager.isDebug() && !lifeCycle.startWorker(port)) {
             throw new WorkerInitializationException(
                     ErrorCodes.INTERNAL_SERVER_ERROR_BACKEND_WORKER_INSTANTIATION);
         }
         String modelName = model.getModelName();
-        listener.notifyChangeState(modelName, WorkerStateListener.WORKER_STARTED);
+        setState(WorkerState.WORKER_STARTED);
```

Comment:
Why did we need to make these changes?

Reply:
We can keep tracking the state in the worker.

Reply:
Ok. Makes sense. I was making similar changes, was curious why you were making this change or where you were using this.